File: nomad-samples/js/media-api/media/content-search.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function contentSearch(AUTH_TOKEN, FILTERS, RETURNED_FIELD_NAME, FIELD_NAME, SORT_TYPE) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
    const BODY = {  
        filters: FILTERS,
        returnedFieldNames: RETURNED_FIELD_NAME,
        sortFields: [  
            {  
                fieldName: FIELD_NAME,  
                sortType: SORT_TYPE 
            }  
        ]  
    };

    // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/portal/search`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    apiExceptionHandler(RESPONSE, "Content search failed");
}

File: nomad-samples/python/Media API/media/content_search.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def content_search(AUTH_TOKEN: str) -> dict:

    # Check for valid parameters
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")

    API_URL = PORTAL_API_URL + "/portal/search"
        
    # Create header for the request
    HEADERS = {
  	    "Authorization": "Bearer " + AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    # Build the payload BODY
    BODY = {  
        "filters": [  
            {  
                "fieldName": "contentDefinitionId",  
                "operator": "Equals",  
                "values": "d34f116d-2a51-4d4a-b928-5dd581d9fd5e"  
            }  
        ],  
        "returnedFieldNames": [  
            "title"  
        ],  
        "sortFields": [  
            {  
                "fieldName": "title",  
                "sortType": "Ascending"  
            }  
        ]  
    }

    try:
        # Send the request
        RESPONSE = requests.post(API_URL, headers= HEADERS, data= json.dumps(BODY))

        if RESPONSE.status_code != 200:
            raise Exception("Response returned " + str(RESPONSE.status_code))

        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "Content searching failed")

